cmd/assetsvc/postgres_db_test.go

```diff
@@ -0,0 +1,221 @@
+		},
+		{
+			name: "it returns the chart matching the chartid",
+			existingCharts: map[string][]models.Chart{
+				"namespace-1": []models.Chart{
+					models.Chart{ID: "chart-1", Name: "my-chart"},
+				},
+			},
+			chartId:       "chart-1",
+			namespace:     "namespace-1",
+			expectedErr:   nil,
+			expectedChart: "my-chart",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			pam, cleanup := getInitializedManager(t)
+			defer cleanup()
+			for namespace, charts := range tc.existingCharts {
+				insertCharts(t, pam, charts, models.Repo{Name: repoName, Namespace: namespace})
+			}
+
+			chart, err := pam.getChart(tc.chartId)
+
+			if got, want := err, tc.expectedErr; got != want {
+				t.Fatalf("got: %+v, want: %+v", got, want)
+			}
+			if got, want := chart.Name, tc.expectedChart; got != want {
+				t.Errorf("got: %q, want: %q", got, want)
+			}
+		})
+	}
+}
+
+func TestGetVersion(t *testing.T) {
+	dbutilstest.SkipIfNoPostgres(t)
+	const repoName = "repo-name"
+
+	testCases := []struct {
+		name string
+		// existingCharts is a map of charts per namespace
+		existingCharts   map[string][]models.Chart
+		chartId          string
+		namespace        string
+		requestedVersion string
+		expectedVersion  string
+		expectedErr      error
+	}{
+		{
+			name:        "it returns an error if the chart does not exist",
+			chartId:     "doesnt-exist-1",
+			namespace:   "doesnt-exist",
+			expectedErr: sql.ErrNoRows,
+		},
+		{
+			name: "it does not ?! return an error if the chart version does not exist",
+			existingCharts: map[string][]models.Chart{
+				"namespace-1": []models.Chart{
+					models.Chart{ID: "chart-1", ChartVersions: []models.ChartVersion{
+						models.ChartVersion{Version: "1.2.3"},
+					}},
+				},
+			},
+			chartId:          "chart-1",
+			namespace:        "namespace-1",
+			requestedVersion: "doesnt-exist",
+			expectedVersion:  "1.2.3",
+		},
+		{
+			name: "it returns the chart version matching the chartid and version",
+			existingCharts: map[string][]models.Chart{
+				"namespace-1": []models.Chart{
+					models.Chart{ID: "chart-1", ChartVersions: []models.ChartVersion{
+						models.ChartVersion{Version: "1.2.3"},
+					}},
+				},
+			},
+			chartId:          "chart-1",
+			namespace:        "namespace-1",
+			requestedVersion: "1.2.3",
+			expectedVersion:  "1.2.3",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			pam, cleanup := getInitializedManager(t)
+			defer cleanup()
+			for namespace, charts := range tc.existingCharts {
+				insertCharts(t, pam, charts, models.Repo{Name: repoName, Namespace: namespace})
+			}
+
+			chart, err := pam.getChartVersion(tc.chartId, tc.requestedVersion)
+
+			if got, want := err, tc.expectedErr; got != want {
+				t.Fatalf("got: %+v, want: %+v", got, want)
+			}
+			if tc.expectedErr != nil {
+				return
+			}
+			// The function just returns the chart with only the one version?
```

Comment:
answering the comment yes, this function is meant to return a single chartVersion

Reply:
The bit that is confusing me is chartVersion == chart with only one version (as opposed to some other object). (removed the '?')

Reply:
yeah, not the most trivial thing. This comes from the mongodb implementation in which when you request a ChartVersion the query filtered all but the matching version (still not perfect but easier to read)